### PR TITLE
MB-1815 issue fixed.

### PR DESCRIPTION
**Pull request by indikasampath2000**
Non durable topic subscriber connection closed through subscription list page. The subscriber connection closed without issue. But topic couldn't delete in the topic list page due to registry resource not getting cleared. The issue was user subject not passed to the authorization when closing the connection. This was fixed by changing the close connection and delete registry entry as two different calls.

Public JIRA - https://wso2.org/jira/browse/MB-1815

### CODE REVIEW

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java [hunk 1]
```diff
@@ -214,44 +213,48 @@ public void deleteQueue(String queueName) throws QueueManagerException {
...
-                    //get admin role of admin user (super tenant admin or tenant admin)
-                    String[] userRoles = userRealm.getUserStoreManager().getRoleListOfUser(userName);
-                    String adminRole = userRealm.getRealmConfiguration().getAdminRoleName();
-                    String role = null;
-                    for (String userRole : userRoles) {
-                        if (userRole.equals(adminRole)) {
-                            role = userRole;
-                            break;
-                        }
+            } else if (subscriptionId.startsWith("tmp_")) {
```
Comment (a5anka) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java]: MQTT protocol queue names might not start with "tmp_". I think we need to do depending on the protocol type.

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp [hunk 2]
```diff
@@ -96,17 +96,21 @@
 
     function closeSubscription(obj) {
         var aTag = jQuery(obj);
+        var isDurable = aTag.attr('isDurable');
         var subscriptionID = aTag.attr('subscription-id');
-        var subscriptionDestination = aTag.attr('subscription-destination');
+        var subscribedQueueOrTopicName = aTag.attr('subscribedQueueOrTopicName');
```
Comment (a5anka) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp]: Isn't destination is the better name than QueueOrTopicName?

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java (hunk 1 above)
Comment (indikasampath2000) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java]: Only AMQP topic subscription details stored in the registry. MQTT topic subscription details not stored in the registry. Therefore, this won't become a problem.

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp (hunk 2 above)
Comment (indikasampath2000) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp]: I just change name to comply with org.wso2.carbon.andes.core.types.Subscription#getSubscribedQueueOrTopicName ;)

**Pull request merged**